cylc lint non zero code from warnings

ColemanTom opened the pull request:
Closes https://github.com/cylc/cylc-flow/issues/5545

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.

~No tests - I didn't think they were needed for this change.~

I have not updated docs as I didn't see where it would be appropriate to add anything about this.

Targetting the 8.1.x branch, but can change if required.

oliver-sanders commented on cylc/flow/scripts/lint.py:
```diff
@@ -663,6 +664,13 @@ def get_option_parser() -> COP:
+        '--error-on-issues', '-e',
+        help='Return an error code if any issues are raised.',
+        action='store_true',
+        default=False,
```
I think this is a sensible default:

```suggestion
        default=True,
```

ColemanTom replied:
Ok, thanks for pointing me at the tests and also the suggestion about `--exit-zero`.

ColemanTom replied:
Changed to `--exit-zero`, default of False is being kept because of this. Other comments have been taken into account.

oliver-sanders commented:
Not a bugfix but small and isolated enough to go into 8.1.5.

hjoliver commented:
(I agree, this should be the default behaviour)

ColemanTom commented:
Ok. Updated

* return non-0 by default. 
* --exit-zero flag added
* functional tests updated and a few new ones added
* renamed some test case names to be more descriptive (and one of them was a duplicate name)
* changed `exit()` to `sys.exit()` as exit is meant for interactive shells, not for scripts, and I was modifying the lines anyway (see the [documentation on this here](https://docs.python.org/3/library/constants.html#constants-added-by-the-site-module))

hjoliver commented:
> changed exit() to sys.exit() as exit is meant for interactive shells, not for script

Huh, I was not even aware of (non-`sys`) `exit()` !

hjoliver reviewed (approved):
👍

oliver-sanders commented:
> Huh, I was not even aware of (non-sys) exit() !

It's how you get out of interactive sessions, not sure how that got into the code.

oliver-sanders reviewed (approved):
:+1:

hjoliver commented:
> It's how you get out of interactive sessions, not sure how that got into the code.

It's not how I get out (Ctrl-D).
